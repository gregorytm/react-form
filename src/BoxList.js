import React, { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

function BoxList() {
  const [boxes, setBoxes] = useState([]);
  const addBox = (userBox) => {
    setBoxes((boxes) => [...boxes, userBox]);
  };
  const remove = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  const boxProperties = boxes.map((box) => (
    <Box
      key={box.id}
      id={box.id}
      width={box.width}
      height={box.height}
      color={box.color}
      handleRemove={remove}
    />
  ));

  return (
    <div>
      <NewBoxForm createBox={addBox} />
      {boxProperties}
    </div>
  );
}

export default BoxList;
