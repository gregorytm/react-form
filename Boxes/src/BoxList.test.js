import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snaqpshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});
