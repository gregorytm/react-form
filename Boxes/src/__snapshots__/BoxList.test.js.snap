// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snaqpshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <div>
          <label
            for="height"
          >
            height
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="width"
          >
            width
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="color"
          >
            color
          </label>
          <input
            id="color"
            name="color"
            type="text"
            value=""
          />
        </div>
        <button
          id="oneButton"
        >
          Add box
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
